bench/lib/parameters.js

```diff
@@ -0,0 +1,33 @@
+const isStyleBench = process.env.STYLE_BENCHMARK;
```

Comment:
Why do we need two separate "isStyleBench" flags set? (`process.env.STYLE_BENCHMARK` and also `?style_bench=true`) Shouldn't one of them be enough?

Reply:
Basically, because this flag needs to be available both in `index.html` and all the benchmark `.js` files. You can see in `index.html` that we need to somehow determine the type of test in order to know how to set the tests up. The only way I could think to make this available in a script tag on an html page was to put it in the URL. I recognized this redundancy and I hate it, so I will gladly change it if we can think of a simpler way to make the test type parameter available in both places.

https://github.com/mapbox/mapbox-gl-js/blob/69335765d0f125f56f9f7ca8484ea903b70d992f/bench/index.html#L20

Reply:
Hmm yeah I see the challenge. I guess you _could_ wait for the local script to resolve and then look up the `isStyleBench` used there, but that would be a weird timing dependency? Or you could make index.html into something templated and have the watch script populate the parameter accordingly? It's probably not too big a deal if it's hard...

Reply:
This is one of those things that seems like a huge issue when you've been deep in the weeds for a couple weeks. Then you take a step back and realize the solution was obvious. Just using the URL parameter works fine and simplifies the API considerably. D'oh!

Reply:
It looks like there's more divergent branches than shared code in some of these files, so rather than trying to share all the same files between the two benchmarking modes, what about an approach where we use different paths/`index.html`/`benchmarks.js` files?

For example, we could have the version-comparing benchmarks at `/bench/versions/` and the style-comparing benchmarks at `/bench/styles/`.

Where we do want to share code between the modes, we can do that by extracting components or functions and importing them into both.